.github/workflows/hadolint-ci.yml

```diff
@@ -0,0 +1,23 @@
+name: 'Hadolint Linting'
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: hadolint/hadolint-action@v3.1.0
+      with:
+        dockerfile: "Dockerfile*"
+        recursive: true
+        # Ignore https://github.com/hadolint/hadolint/wiki/DL3041
+        # Ignore https://github.com/hadolint/hadolint/wiki/DL3028
+        # Ignore https://github.com/hadolint/hadolint/wiki/DL3050
```

Comment:
Could you write a comment describing why we have these lint exceptions?

Reply:
Would you like this comment in the code or here? However, the wiki links in the comments detail why we had to ignore those errors.

Reply:
I don't see anything in the wiki about why we are ignoring these lints.

Reply:
Since we are using `gem` and `dnf` to install the packages, for the error to be fixed, we need to hardcode the preferred version number to each package. But in order to always have the updated version, we avoid hardcoding the numbers and ignore the error (this is for the first two ignores)

Also for the last ignore process, we cannot add a label at the beginning of the docker file cos that feature has been deprecated for Security reason, however since the repo is open-source, we can add label and run the ignore.

Reply:
It does not matter if we use `gem` and `dnf`, even `apt` has the same issues. But this is AFAIK tho, you can check the wiki for more information on my context.

Reply:
Well, following @LesnyRumcajs review, I'd have to pin the versions and remove the ignore so going ahead with that.

Reply:
@samuelarogbonlo they don't have the same issues. You can expect that a version x.y.z of a particular dependency is *always* available from https://rubygems.org/ (unless it gets explicitly yanked). You cannot make such an assumption about dnf and apt packages. From what I know, at least for Alpine, once a version x.y.z+1 is released, the dependency x.y.z is dropped. I recall the same thing for Debian packages, maybe with a longer grace period.

Reply:
I understand and I think I didn't explain it well. This is what I mean; basically, I never meant to say it was an issue. I only said that we ignored the failing hadolint ci error because 

- firstly the original dockerfiles didn’t have versions pinned to them, pinning versions is actually a best practice but we ignored it because the original dockerfile didn’t have them and also we dont know which exact versions to set.
- The same issue will happen if we are using `apt` and `dnf` and we don't pin versions to them. The wiki is [here](https://github.com/hadolint/hadolint/wiki/DL3028) for ruby - here, you can see that we have to pin the version to resolve the failing CI issues. Now for `apt', check [here](https://github.com/hadolint/hadolint/wiki/DL3008) and see what I mean as well. 

As I said, this is AFAIK, but I might be missing a piece of information or wrongly understanding things.